docs/resource/cicd.md

```diff
@@ -0,0 +1,195 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: CICD
+--->
+
+# CICD
+
+<!-- toc -->
+
+- [Continuous Integration and Continuous Deployment (CICD)](#continuous-integration-and-continuous-deployment-cicd)
+  - [CICD pipeline resource attributes](#cicd-pipeline-resource-attributes)
+  - [CICD pipeline run resource attributes](#cicd-pipeline-run-resource-attributes)
+  - [CICD worker resource attributes](#cicd-worker-resource-attributes)
+- [Version Control System (VCS)](#version-control-system-vcs)
+  - [VCS repository resource attributes](#vcs-repository-resource-attributes)
+  - [VCS reference resource attributes](#vcs-reference-resource-attributes)
+
+<!-- tocstop -->
+
+See also:
+
+- The [Trace semantic conventions for CICD](/docs/cicd/cicd-spans.md)
+
+## Continuous Integration and Continuous Deployment (CICD)
+
+CICD aims to increase early defect discovery, increase productivity, and provide faster release cycles through automating builds, tests and deployment of software.
+
+### CICD pipeline
+
+<!-- semconv resource.cicd.pipeline -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+
+**Status:** ![Development](https://img.shields.io/badge/-development-blue)
+
+**type:** `cicd.pipeline`
+
+**Description:** A pipeline is a series of automated steps that helps software teams deliver code.
+
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| [`cicd.pipeline.name`](/docs/attributes-registry/cicd.md) | string | The human readable name of the pipeline within a CI/CD system. | `Build and Test`; `Lint`; `Deploy Go Project`; `deploy_to_environment` | `Recommended` | ![Development](https://img.shields.io/badge/-development-blue) |
+
+<!-- markdownlint-restore -->
+<!-- prettier-ignore-end -->
+<!-- END AUTOGENERATED TEXT -->
+<!-- endsemconv -->
+
+### CICD pipeline run
+
+To efficiently set the following attributes for all spans of a pipeline run they should be defined as the following resource.
+
+Since this resource is unique for each pipeline run, this means instantiating separate [TracerProviders](https://opentelemetry.io/docs/specs/otel/trace/api/#tracerprovider) for each pipeline run in order to be able to configure the resources of any spans emitted through that tracer.
```

Comment:
Related to https://github.com/open-telemetry/semantic-conventions/issues/1111 https://github.com/open-telemetry/semantic-conventions/issues/1184#issuecomment-2773594630, we should add a warning here that if the `CICD pipeline run` resource is used for metrics, then it might cause issues with high-cardinality.

Reply:
I added a warning in 7f6b5b38b0620ea6ef75cfe24bffa4aa2b95cab7.